undefined reference to `glXGetCurrentContext'

Hello,
We're aiming to compile OSD 3.5.0 on Ubuntu 20 against CUDA 11.7 for a Quadro RTX 4000 GPU. After changing `compute_20` to `compute_75` in `CMakeLists.txt`, we're hitting the following `glX*` related errors, whether GLEW and CUDA are on or off. Of course, we'd like CUDA on.

The error:
```
...
[ 37%] Built target examples_common_gl_obj
[ 38%] Generating mtlPtexCommon.gen.h
[ 39%] Generating glslPtexCommon.gen.h
[ 39%] Generating hlslPtexCommon.gen.h
[ 39%] Building CXX object examples/common/CMakeFiles/examples_common_ptex_obj.dir/ptexMipmapTextureLoader.cpp.o
[ 39%] Building CXX object examples/common/CMakeFiles/examples_common_ptex_obj.dir/glPtexMipmapTexture.cpp.o
[ 39%] Built target examples_common_ptex_obj
[ 39%] Generating shader_gl3.gen.h
[ 39%] Generating shader.gen.h
[ 40%] Building CXX object examples/glViewer/CMakeFiles/glViewer.dir/glViewer.cpp.o
[ 40%] Linking CXX executable ../../bin/glViewer
/usr/bin/ld: ../common/CMakeFiles/examples_common_gl_obj.dir/clDeviceContext.cpp.o: in function `CLDeviceContext::Initialize()':
clDeviceContext.cpp:(.text+0x4ed): undefined reference to `glXGetCurrentContext'
/usr/bin/ld: clDeviceContext.cpp:(.text+0x4f6): undefined reference to `glXGetCurrentDisplay'
/usr/bin/ld: ../common/CMakeFiles/examples_common_gl_obj.dir/cudaDeviceContext.cpp.o: in function `_GetCudaDeviceForCurrentGLContext()':
cudaDeviceContext.cpp:(.text+0x5b): undefined reference to `glXGetCurrentDisplay'
collect2: error: ld returned 1 exit status
make[2]: *** [examples/glViewer/CMakeFiles/glViewer.dir/build.make:156: bin/glViewer] Error 1
make[1]: *** [CMakeFiles/Makefile2:3767: examples/glViewer/CMakeFiles/glViewer.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```
Here's an example of the cmake command we've tried:
```
cmake \
-D CMAKE_CXX_STANDARD=14 \
-D CMAKE_INSTALL_PREFIX=/sw/builds/opensibdiv/3.5.0  \
-D CMAKE_BUILD_WITH_INSTALL_RPATH=True \
-D CMAKE_INSTALL_RPATH_USE_LINK_PATH=True \
-D PTEX_LOCATION=/sw/builds/ptex/2.4.2 \
-D CUDA_SDK_ROOT_DIR=/usr/local/cuda-11.7 \
-D CUDA_TOOLKIT_ROOT_DIR=/usr/local/cuda-11.7 \
-D GLFW_LOCATION=/sw/builds/glfw/3.3.8 \
-D TBB_INCLUDE_DIR=/sw/builds/tbb/2020.3/include \
-D TBB_tbb_LIBRARY=/sw/builds/tbb/2020.3/lib/libtbb.so \
-D NO_CLEW=ON \
-D NO_GLEW=ON \
-D CUDA_cupti_LIBRARY=/usr/local/cuda-11.7/lib64/libcupti.so \
-D CUDA_CUDA_LIBRARY=/usr/local/cuda-11.7/lib64/stubs/libcuda.so \
-D NO_MACOS_FRAMEWORK=ON \
-D NO_METAL=ON \
-D _OPENCL_CPP_INCLUDE_DIRS=/usr/local/cuda-11.7/include \
-D NO_REGRESSION=ON \
..
```

Thank you for having a look,
- RGBA Research

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -803,6 +803,11 @@ macro(osd_add_possibly_cuda_executable target folder)
 
     set_target_properties(${target} PROPERTIES FOLDER ${folder})
 
+    # Workaround link dependencies for cuda examples on platforms with GLX
+    if(CUDA_FOUND AND OpenGL_GLX_FOUND)
+        target_link_libraries(${target} OpenGL::GLX)
+    endif()
+
     if(CMAKE_COMPILER_IS_ICC)
         target_link_libraries(${target} ${ICC_LIBRARIES})
     endif()

